src/model.rs

```diff
@@ -409,3 +409,197 @@ macro_rules! jsonapi_model {
         }
     );
 }
+
+/// A trait for any struct that can be converted from/into a
+/// [`PartialResource`](api/struct.PartialResource.tml). The only requirement is
+/// that your struct does not have a `id: String` field.
+/// You shouldn't be implementing PartialJsonApiModel manually, look at the
+/// `jsonapi_model!` macro instead.
+pub trait PartialJsonApiModel: Serialize
+where
+    for<'de> Self: Deserialize<'de>,
+{
+    #[doc(hidden)]
+    fn jsonapi_type(&self) -> String;
+    #[doc(hidden)]
+    fn relationship_fields() -> Option<&'static [&'static str]>;
+    #[doc(hidden)]
+    fn build_relationships(&self) -> Option<Relationships>;
+
+    fn from_partial_jsonapi_resource(resource: &PartialResource) -> Result<Self> {
+        Self::from_serializable(Self::partial_resource_to_attrs(resource))
+    }
+
+    fn from_jsonapi_document(doc: &DocumentData) -> Result<Self> {
+        match doc.data.as_ref() {
+            Some(primary_data) => match *primary_data {
+                PrimaryData::None => bail!("Document had no data"),
+                PrimaryData::Single(_) => unimplemented!(),
```

Comment:
This needs an implementation - presumably you should be able to build a partial resource from a full document, though not the other way around.